File: socialNetwork/src/main/java/com/ftn/socialNetwork/model/entity/FriendRequest.java
package com.ftn.socialNetwork.model.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDateTime;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "friendRequests")
public class FriendRequest {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private boolean approved;

    @Column(nullable = false, columnDefinition = "CREATEDAT DEFAULT CURRENT_CREATEDAT")
    private LocalDateTime created_at;
    @Column(nullable = false)
    private LocalDateTime at;
    @ManyToOne
    private User user;
}

File: src/app/models/friendRequest.ts
import { User } from "./user";

export class FriendRequest {
    id: number;
    approved: boolean ;
    created_at: Date;
    at: Date | null;
    fromUser: User;
    forUser: User;

  constructor(id: number,  approved: boolean ,created_at: Date, at: Date | null, fromUser: User, forUser: User
  ) { 
    this.id = id;
    this.approved = approved;
    this.created_at = created_at;
    this.at = at;
    this.fromUser = fromUser;
    this.forUser = forUser;

}
}